src/EnergyPlus/MixedAir.cc

```diff
@@ -3715,63 +3670,37 @@ Real64 VentilationMechanicalProps::CalcMechVentController(EnergyPlusData &state,
-                ZoneOACalc[static_cast<int>(DataSizing::OAFlowCalcMethod::PerArea)] = curZone.FloorArea * thisMechVentZone.ZoneOAAreaRate;
-                ZoneOACalc[static_cast<int>(DataSizing::OAFlowCalcMethod::PerZone)] = thisMechVentZone.ZoneOAFlowRate;
-                ZoneOACalc[static_cast<int>(DataSizing::OAFlowCalcMethod::ACH)] = (thisMechVentZone.ZoneOAACHRate * curZone.Volume) / 3600.0;
-
-                // Calc the breathing-zone OA flow rate
-                int OAIndex = thisMechVentZone.ZoneDesignSpecOAObjIndex; // index to design specification outdoor air objects
-                if (OAIndex > 0) {
-                    switch (state.dataSize->OARequirements(OAIndex).OAFlowMethod) {
-                    case DataSizing::OAFlowCalcMethod::Sum: {
-                        ZoneOABZ = multiplier * (ZoneOACalc[static_cast<int>(DataSizing::OAFlowCalcMethod::PerPerson)] +
-                                                 ZoneOACalc[static_cast<int>(DataSizing::OAFlowCalcMethod::PerArea)] +
-                                                 ZoneOACalc[static_cast<int>(DataSizing::OAFlowCalcMethod::PerZone)] +
-                                                 ZoneOACalc[static_cast<int>(DataSizing::OAFlowCalcMethod::ACH)]);
-                    } break;
-                    case DataSizing::OAFlowCalcMethod::Max: {
-                        ZoneOABZ = multiplier * max(ZoneOACalc[static_cast<int>(DataSizing::OAFlowCalcMethod::PerPerson)],
-                                                    ZoneOACalc[static_cast<int>(DataSizing::OAFlowCalcMethod::PerArea)],
-                                                    ZoneOACalc[static_cast<int>(DataSizing::OAFlowCalcMethod::PerZone)],
-                                                    ZoneOACalc[static_cast<int>(DataSizing::OAFlowCalcMethod::ACH)]);
-
-                    } break;
-                    default: {
-                        ZoneOABZ = multiplier * ZoneOACalc[static_cast<int>(state.dataSize->OARequirements(OAIndex).OAFlowMethod)];
-                        break;
-                    }
-                    }
-                } else {
-                    ZoneOABZ = multiplier * ZoneOACalc[static_cast<int>(DataSizing::OAFlowCalcMethod::PerPerson)];
-                }
+                int const ZoneNum = thisMechVentZone.zoneNum;
+
+                // Use calcDesignSpecificationOutdoorAir function to support simple DSOA and DSOA:SpaceList
+                bool const useOASch = true;
+                bool const perPersonNotSet = false;  // placeholder function argument
+                bool const maxOAVolFlowFlag = false; // placeholder function argument
+                int constexpr spaceNum = 0;          // placeholder function argument
+                bool const calcIAQMethods = false;   // If the zone or space OA method is IAQProcedure, PCOccSch or PCDesOcc then return zero
+                thisMechVentZone.zoneOABZ = DataSizing::calcDesignSpecificationOutdoorAir(state,
```

Comment:
. . . with a call to `calcDesignSpecificationOutdoorAir`, and save `zoneOABZ` for later use.